accepted/2020/platform-exclusion/platform-exclusion.md

```diff
@@ -0,0 +1,217 @@
+
+1. The so marked APIs are now considered platform-specific, which means that
+   just calling the API from portable code will now always be flagged by the
+   analyzer.
+2. Every time .NET adds another operating system all call sites guarding calls
+   to these APIs now also have to check for this new OS.
+
+In this spec, we're proposing a mechanism by which we can mark APIs as
+unsupported.
+
+## Scenarios and User Experience
+
+* Building a Blazor WebAssembly app
+    - Using an API that is unsupported will be flagged by the analyzer.
+* Building a class library
+    - Building a regular class library (targeting `netstandard` or `net5.0`)
+      will not complain about APIs that are unsupported in Blazor.
+* Building a class library for use in Blazor WebAssembly
+    - Developer can modify the project file to explicitly indicate that they
+      want to see issues with APIs that are unsupported in Blazor.
+
+## Requirements
+
+### Goals
+
+* Be able to mark APIs that are unsupported on a particular OS
+* Allows platforms to be considered supported by default with the ability to add
+  or remove platforms from the project file and SDK.
+* Blazor WebAssembly is only considered supported when building a Blazor
+  WebAssembly app. For regular class libraries (`netstandard` or `net5.0`) the
+  default assumes that it's not supported and thus won't generate any warnings.
+  However, the developer must still be able to manually include Blazor
+  WebAssembly manually.
+* The annotations for unsupported APIs can express that an API was only
+  unsupported for a specific version.
+
+### Non-Goals
+
+## Design
+
+### Attribute
+
+```C#
+namespace System.Runtime.Versioning
+{
+    // Rename RemovedInOSPlatformAttribute to
+    // UnsupportedOSPlatformAttribute
+    [AttributeUsage(AttributeTargets.Assembly |
+                    AttributeTargets.Class |
+                    AttributeTargets.Constructor |
+                    AttributeTargets.Event |
+                    AttributeTargets.Method |
+                    AttributeTargets.Module |
+                    AttributeTargets.Property |
+                    AttributeTargets.Struct,
+                    AllowMultiple=true, Inherited=false)]
+    public sealed class UnsupportedOSPlatformAttribute : OSPlatformAttribute
+    {
+        public UnsupportedOSPlatformAttribute(string platformName);
+    }
+```
+
+The semantics of this attribute are as follows:
+
+* Marks an assembly, module, or API as unsupported on a given OS platform.
+* The `platformName` can contain no version number, a single version number, or
+  a range:
+    - `browser`. Marks the API as unsupported on all version of `browser`. This
+      will be used for APIs that can't be supported on a particular OS platform.
+    - `ios14.0`. Marks the API as unsupported starting with iOS 14.0. This will
+      be used for OS platform APIs that were removed/disabled.
+    - `windows0.0-10.0.19041`. Marks the API as unsupported for 10.0.19041 and
```

Comment:
If ranges are to be supported, I recommend git-style .. instead of a hyphen-minus.  Hyphens seem like they're more likely to appear in an OS name than a .. would be.